### DISCUSSION TITLE
Versions got dropped to the initial one

### DISCUSSION THREAD

**Discussion by morozovamv** [jscutlery/semver#759] [Q&A] [answered]
Hi! I've upgraded my project to nx@17.2.6 and after running version update on GitLab ci `yarn exec nx run-many --all --target=version --skip-nx-cache` it ran `nx run ***:version` for each of the app/library and dropped all versions to 0.1.0. 
Normally that command would skip libs that didn't have any changes and it would upgrade version for example 39.9.1 -> 39.9.2.
Do you know why this could happen?

We use "@jscutlery/semver": "4.1.0".

**Comment by edbzn** [comment]
Hi @morozovamv, can you show me the semver config you are using?

**Reply by morozovamv** [reply]
Yes, sure
project.json
```
"version": {
  "executor": "@jscutlery/semver:version",
  "options": {
    "preset": "conventional",
    "push": false,
    "noVerify": true,
    "trackDeps": true,
    "tagPrefix": "${projectName}@",
    "commitMessageFormat": "chore(${projectName}): Release version ${version}"
  }
}
```

**Reply by edbzn** [reply]
Could you try removing the $ sign in the interpolation? So it should look like `"{projectName}@"`.

**Reply by morozovamv** [reply]
I made that change and reverted back the version I had before in my library
<img width="999" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/e5eb840f-8198-448f-97dd-fc7c7e50ce41">
Unfortunately wrong version was calculated again
<img width="873" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/61e92f46-187a-4516-b4cd-ea39c4ed6ca9">

**Reply by edbzn** [reply]
I suspect a version conflict between transitive deps, are you using commitlint? If yes, you could you give me the output of those two commands? 

```
yarn why conventional-changelog-conventionalcommits
yarn why conventional-changelog-angular
```

**Reply by morozovamv** [reply]
Yes, we do
<img width="814" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/a17c87d4-ab50-4802-8cca-65dda99331dd">

**Reply by edbzn** [reply]
Looks right, and what do you get with `yarn why git-semver-tags`?

**Reply by morozovamv** [reply]
Seems ok too
<img width="559" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/72880f27-e298-43f4-83c9-ab81d240f844">

Just to give you more context, these are deps changes, maybe you notice something
<img width="1002" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/3ebe8655-798c-43e1-9c0e-42c2c72ef1be">
<img width="997" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/92bfbb20-5e6a-42eb-8418-1a5b3258ff34">
<img width="929" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/00ac764a-c9f8-4a91-a9bf-330a04567361">

**Reply by morozovamv** [reply]
More context: there is another issue with commitlint (https://github.com/conventional-changelog/commitlint/issues/3820), I also made a workaround that was suggested in the issue. I'm not sure if that is related to my current issue.

**Reply by edbzn** [reply]
Not sure it's related actually. I'm investigating I'll keep you informed.

**Reply by morozovamv** [reply, marked as answer]
Seem to be fixed, steps I've done:
- deleted all wrong commits from git history (ex. "chore(**$**expereo-ui): Release version 0.0.2"; $-sign shouldn't be there)
- deleted all wrong tags (like **$**expereo-ui@0.0.2; $-sign shouldn't be there)
- moved our local tags to the latest commit
- changed config
<img width="1440" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/c8dad19b-3619-4233-8f9d-d40cc67c7cc7">
As a result version was generated correctly on GitLab CI
<img width="864" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/47ec1154-5d67-40e1-b32e-55b57a96add2">

I don't know at what point it got broken, hopefully future MR will be also fine.

**Reply by morozovamv** [reply]
I also noticed another issue with versions on the changelog.md, it was wrong since the beginning and after I've done the versions fix, it changed but it is still wrong. Could you point me, what can I check to fix it?
The issue is that in the changelog we track version updates. Previously after some dependency got an update, there was a message about it, but it had wrong dependency version ("dep ***" updated to version "0.1.0") - it was always 0.1.0. Now, after fix, instead of 0.1.0 there is a version of a parent dependency, not of the dependency itself. You can see it on the screenshot. Setup is done according to (https://github.com/jscutlery/semver?tab=readme-ov-file#tracking-dependencies)
<img width="1447" alt="image" src="https://github.com/jscutlery/semver/assets/19936650/84845bc6-5bd4-4d6a-9bf3-355777ce19ff">

**Reply by edbzn** [reply]
I think it's related to a known issue: #526

**Reply by morozovamv** [reply]
All clear, thank you for your help @edbzn!
